File: web/src/types/humanTask.ts
// 人工任务状态（简化版）
export type HumanTaskStatus = 'pending' | 'completed' | 'cancelled';

// 人工任务（简化版）
export interface HumanTask {
  id: string;
  threadId: string;
  invocationId: string;      // 关联的 invocation ID
  agentConfigId: string;     // Agent 角色配置 ID
  agentName: string;         // Agent 名称
  waitReason: string;        // 等待原因
  projectId: string;         // 项目 ID
  projectName: string;       // 项目名称
  threadName: string;        // 任务名称（Thread 名称）
  status: HumanTaskStatus;
  createdAt: string;
  completedAt?: string;      // 完成时间
}

// 提交人工任务请求（简化版）
export interface SubmitHumanTaskRequest {
  outputContent: string;
  outputFiles?: string[];
}

// 提交人工任务响应（简化版）
export interface SubmitHumanTaskResponse {
  success: boolean;
  triggered: boolean;
}

File: internal/model/human_task.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// ========== Human Task Models ==========

// HumanTaskStatus 人工任务状态
type HumanTaskStatus string

const (
	HumanTaskStatusPending   HumanTaskStatus = "pending"
	HumanTaskStatusCompleted HumanTaskStatus = "completed"
	HumanTaskStatusCancelled HumanTaskStatus = "cancelled"
)

// HumanTask 人工任务（简化版）
type HumanTask struct {
	ID            uuid.UUID       `json:"id"`
	ThreadID      uuid.UUID       `json:"threadId"`
	InvocationID  uuid.UUID       `json:"invocationId"`   // 关联 invocation
	AgentConfigID uuid.UUID       `json:"agentConfigId"`  // 等待用户的 Agent
	AgentName     string          `json:"agentName"`      // Agent 名称
	WaitReason    string          `json:"waitReason"`     // 等待原因（输出摘要）
	ProjectID     uuid.UUID       `json:"projectId"`      // 项目 ID
	ProjectName   string          `json:"projectName"`    // 项目名称
	ThreadName    string          `json:"threadName"`     // 任务名称（Thread 名称）
	Status        HumanTaskStatus `json:"status"`
	CreatedAt     time.Time       `json:"createdAt"`
	CompletedAt   *time.Time      `json:"completedAt"`
}

func (t *HumanTask) TableName() string {
	return "human_tasks"
}

// SubmitHumanTaskRequest 提交交付物请求
type SubmitHumanTaskRequest struct {
	OutputContent string   `json:"outputContent"`
	OutputFiles   []string `json:"outputFiles"`
}

// SubmitHumanTaskResponse 提交响应
type SubmitHumanTaskResponse struct {
	Success   bool           `json:"success"`
	NextAgent *NextAgentInfo `json:"nextAgent,omitempty"`
	Triggered bool           `json:"triggered"`
}

// NextAgentInfo 下游 Agent 信息
type NextAgentInfo struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}